### ISSUE TITLE
[EVM] Add support for abi.encode() and abi.decode() to Cadence 

### ISSUE THREAD

**Issue by franklywatson** [onflow/cadence#2925]
### Issue to be solved

The Axelar cross-chain integration work needs the Ethereum ABI.encode() and ABI.decode() functionality made available, ideally as a native Cadence language function owing to the difficulty of implementing the same logic directly in Cadence code due to the byte level handling. This need will doubtless also arise for other aspects of EVM integration in future since ABIs are very common. 

The reference example in https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/abi.go could in theory be copy-pasted along with tests and then surface language native methods to invoke that logic.

### Suggested Solution

This task is to confirm the feasibility of this, and confirm the naming and spec for the Cadence native functions that would allow Cadence code to run the ABI.encode() and ABI.decode() through them. 

** Definition of done **

* Cadence native language functions for ABI.encode() and ABO.decode() are available in the Cadence runtime
* Tests exist which assert that encode/decode functionality works identically to the Ethereum version
* Tests exist to assert the integration of the functionality in Cadence

**Comment by bluesign** [comment]
I think place for this is in flow-go:  https://github.com/onflow/flow-go/blob/643b532c77eacc4d8b34232afaa8c7416a1cd912/fvm/evm/stdlib/contract.cdc

should be something like `evm.encodeABI` and `evm.decodeABI` ( though @turbolent and @ramtinms knows better )

**Comment by turbolent** [comment]
Yes, this should be implemented in the `EVM` contract defined in flow-go, this is not a part of Cadence. Moving
